plugins/modules/aci_endpoint_ip_tag.py

```diff
@@ -0,0 +1,328 @@
+            "attributes": {
+                "descr": "Production environment",
+                "name": "production"
+            }
+        }
+    }
+filter_string:
+  description: The filter string used for the request
+  returned: failure or debug
+  type: str
+  sample: ?rsp-prop-include=config-only
+method:
+  description: The HTTP method used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: POST
+response:
+  description: The HTTP response from the APIC
+  returned: failure or debug
+  type: str
+  sample: OK (30 bytes)
+status:
+  description: The HTTP status from the APIC
+  returned: failure or debug
+  type: int
+  sample: 200
+url:
+  description: The HTTP url used for the request to the APIC
+  returned: failure or debug
+  type: str
+  sample: https://10.11.12.13/api/mo/uni/tn-production.json
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.cisco.aci.plugins.module_utils.aci import ACIModule, aci_argument_spec, aci_annotation_spec, aci_owner_spec
+
+
+def main():
+    argument_spec = aci_argument_spec()
+    argument_spec.update(aci_annotation_spec())
+    argument_spec.update(aci_owner_spec())
+    argument_spec.update(
+        tenant=dict(type="str"),
+        name=dict(type="str"),
+        endpoint_ip_address=dict(type="str", aliases=["ip_addr", "ip"]),
+        vrf=dict(type="str", aliases=["vrf_name"]),
+        name_alias=dict(type="str"),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "vrf", "endpoint_ip_address"]],
+            ["state", "present", ["tenant", "vrf", "endpoint_ip_address"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    tenant = module.params.get("tenant")
+    endpoint_ip_address = module.params.get("endpoint_ip_address")
+    vrf = module.params.get("vrf")
+    name = module.params.get("name")
+    name_alias = module.params.get("name_alias")
+    annotation = module.params.get("annotation")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+
+    endpoint_ip_tag_rn = None
+    endpoint_ip_tag_module_object = None
+    if endpoint_ip_address and vrf:
+        endpoint_ip_tag_rn = "eptags/epiptag-[{0}]-{1}".format(endpoint_ip_address, vrf)
+        endpoint_ip_tag_module_object = "[{0}]-{1}".format(endpoint_ip_address, vrf)
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="fvEpIpTag",
+            aci_rn=endpoint_ip_tag_rn,
+            module_object=endpoint_ip_tag_module_object,
+            target_filter=dict(ip=endpoint_ip_address, ctxName=vrf),
```

Comment:
Should we add `name` here so we can also query by name?
```suggestion
            target_filter=dict(ip=endpoint_ip_address, ctxName=vrf, name=name),
```